lib/modules/manager/puppet/common.ts

```diff
@@ -0,0 +1,52 @@
+import { regEx } from '../../../util/regex';
+import { parseUrl } from '../../../util/url';
+import type { PackageDependency } from '../types';
+
+export const RE_REPOSITORY_GENERIC_GIT_SSH_FORMAT =
+  regEx(/^(?:git@[^:]*):(.+)$/);
+
+export function parseGitOwnerRepo(
+  git: string,
+  githubUrl: boolean
+): string | PackageDependency {
+  const genericGitSsh = RE_REPOSITORY_GENERIC_GIT_SSH_FORMAT.exec(git);
+
+  if (genericGitSsh) {
+    return genericGitSsh[1].replace(regEx(/\.git$/), '');
+  } else {
+    if (githubUrl) {
+      return git
+        .replace(regEx(/^github:/), '')
+        .replace(regEx(/^git\+/), '')
+        .replace(regEx(/^https:\/\/github\.com\//), '')
+        .replace(regEx(/\.git$/), '');
+    }
+    try {
+      const url = parseUrl(git);
+
+      if (!url) {
+        return invalidUrl(git);
+      }
+
+      return url.pathname
+        .replace(regEx(/\.git$/), '')
+        .replace(regEx(/^\//), '');
+    } catch (err) {
+      return invalidUrl(git);
+    }
+  }
+}
+
+function invalidUrl(sourceUrl: string): PackageDependency {
+  return {
+    gitRef: true,
+    sourceUrl,
+    skipReason: 'invalid-url',
```

Comment:
at the moment we do not support all kinds of git URLs. For example for github we explicitly check for http (missing s), such a url would be "invalid-url" because it would be a high security risk. Other case is when we really cant parse the url, then it is invalid for us to.